packages/contracts-bedrock/interfaces/safe/ITimelockGuard.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.4;
+
+library Enum {
+    type Operation is uint8;
+}
+
+library TimelockGuard {
+    struct ScheduledTransaction {
+        uint256 executionTime;
+        bool cancelled;
+        bool executed;
+    }
```

Comment:
Some context for the fields on the struct:
- We record the execution time, as opposed to the scheduling time, so that the execution time is not dependent on potential changes to the timelock delay which the transaction is scheduled. This results in an easier problem space to reason in.
- Cancelled and Executed are transaction states, as is the implied Pending which happens when `executionTime != 0 && !executed && !cancelled`. We record them to ensure that certain functions can only be called on a transaction when this is in a given state. There would be other ways to record transaction state, such as an `enum TransactionState {NOT_SCHEDULED, SCHEDULED, EXECUTED, CANCELLED}`